### ISSUE TITLE
Need some help

### ISSUE THREAD

**Issue by sant527** [ovh/celery-dyrygent#13]
In celery i have to implement a chord, like situation

![image](https://user-images.githubusercontent.com/6462531/103443937-d607d480-4c89-11eb-8e52-aeda589e9273.png)

 I working on a stock market data. Every hour i have to calculate some value for each stock based on last 100 hours data 
I have 10,000 stocks.

Currently i  am using python multiprocessing. It works fine.

But I am thinking to use celery, As you mentioned celery is slow in this regard in your article. https://www.ovh.com/blog/doing-big-automation-with-celery/

![image](https://user-images.githubusercontent.com/6462531/103443992-7100ae80-4c8a-11eb-815a-cb2971526c14.png)

So i want to use your celery extension. Will this have the same speed as python multiprocessing  utility

**Comment by brabiega** [comment]
Hello!
Assuming that I understand your case correctly.

### tl;dr;
Use native celery chord, if you encounter reliability/scalability problems then try celery-dyrygent.

### Explanation
Multiprcessing will run faster than celery within single host.
If you use celery you can scale beyond single host to process the data faster (e.g 100 workers per host).

Your case seems like a big, *simple* workfllow (Process 10 000 stocks in parallel) which should be fine for a single native celery chord.
So I would try https://docs.celeryproject.org/en/stable/userguide/canvas.html#chords in first place.

celery-dyrygent is an extension to handle *complex* workflows (multiple combination of chains and chords). It can be used for your case, it may improve reliability but it will be slightly slower than native celery chord.

**Comment by brabiega** [comment]
Closing this topic due to inactivity.
